wdl-lint/src/tags.rs

```diff
@@ -32,6 +32,37 @@ pub enum Tag {
     Deprecated,
 }
 
+/// An error for when an unknown tag is encountered.
+#[derive(Debug)]
+pub struct UnknownTagError(String);
+
+impl std::fmt::Display for UnknownTagError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "unknown tag: {}", self.0)
+    }
+}
+
+impl std::error::Error for UnknownTagError {}
+
+impl std::convert::TryFrom<&str> for Tag {
+    type Error = UnknownTagError;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        match value.to_lowercase().as_str() {
```

Comment:
Courtesy of Peter: we can use https://doc.rust-lang.org/stable/std/primitive.str.html#method.eq_ignore_ascii_case to avoid the allocation here